src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -202,6 +213,35 @@ public string GetRelativePath(string filePath)
             return resolvedPath;
         }
 
+        internal static string MapRemotePathToLocal(string remotePath, string connectionString)
+        {
+            string sessionTempPath =
+                Path.Combine(
+                    Path.GetTempPath(),
+                    "PSES-" + Process.GetCurrentProcess().Id,
+                    "RemoteFiles");
+
+            // The path generated by this code will look something like
+            // %TEMP%\PSES-[PID]\RemoteFiles\1205823508\computer-name\MyFile.ps1
```

Comment:
But if you're including the hash of the file's contents in the filename, any change would result in a new hash and hence a new file.  Or maybe I'm misunderstanding the "hash slug".